### ISSUE TITLE
Add AOMap to cockpit

### ISSUE THREAD

**Issue by gilbertohasnofb** [c172p-team/c172p#798]
Create and add AOMap to cockpit, similar to what has been done in the c182. It makes the interior textures look absolutely amazing with it.

AFAIK it's possible to create these using Blender, and then adding the AOMap as a layer in GIMP to the cockpit and interior textures. Maybe Heiko could give us a hand with this.

**Comment by gilbertohasnofb** [comment]
If someone can help me create these AOMaps on Blender, I can then apply them to all interior and cockpit textures of all liveries myself.

**Comment by wlbragg** [comment]
This must be similar to the lightmap creation in Blender that I just learned how to do. I might see if I can do this as well. When creating a lightmap in Blender there is a switch for adding AO to the lightmap. I think that is different than what your looking for here though. Because this is actually added to the texture sheet, not as a lightmap, right?

**Comment by gilbertohasnofb** [comment]
Yes, it would be added as a layer in GIMP and then I'd control the opacity of that layer. If you run into troubles with the AOMap, we can ask Heiko how he did it.

**Comment by wlbragg** [comment]
I think you "bake" the AO map in Blender, I know how to do that, then you combine it with the original texture. I need to know what program he used and the process he used to combine the two ("AO" and "original diffuse texture")?

**Comment by gilbertohasnofb** [comment]
I will point him to this talk then.

**Comment by wlbragg** [comment]
Thanks

**Comment by gilbertohasnofb** [comment]
@wlbragg Perhaps this could be of some help: https://www.youtube.com/watch?v=UbhiZGCAo0Q

(though I tried it myself and failed miserably, when I open our dev files the layout of Blender is completely different than what is the default... :sob: )

**Comment by legoboyvdlp** [comment]
If this were still to be done, I could probably bake the maps... best to wait until #6 snd so on are fixed first perhaps?

**Comment by gilbertohasnofb** [comment]
> best to wait until #6 snd so on are fixed first perhaps?

:+1:

**Comment by legoboyvdlp** [comment]
By the way, seeing you are online, I noticed tigert's link to the OH-CTL gallery is down... you may want to bookmark this:  https://mik.1g.fi/kuvat/Kerhon+kalusto/OH-CTL/

It appears to be the same thing..?

**Comment by gilbertohasnofb** [comment]
@legoboyvdlp Fantastic, lego, thanks for the link! I spent so much time searching for it and could not find. Many thanks.

**Comment by wlbragg** [comment]
@gilbertohasnofb yes, we can give it a shot.

**Comment by gilbertohasnofb** [comment]
Great. No rush, let's finish that PR first then we move on to these other interior issues.

**Comment by gilbertohasnofb** [comment]
@wlbragg If it would be easy to bake these AOmaps for the panel, panel_parts and interior textures I would love to work on them tonight. I finally have a couple of hours free :smile:

**Comment by wlbragg** [comment]
@gilbertohasnofb let me see what I can do. I may run into a time crunch before "your tonight". I'll start on it now and see where it takes me.

**Comment by gilbertohasnofb** [comment]
Ah no rush then. I just thought that if this could just be a matter of a couple of clicks and exporting three png files but I totally understand if you need longer time. I might as well watch some Netflix then :smile:

**Comment by gilbertohasnofb** [comment]
@wlbragg No rush. But we would need them as overlays given that we have different interior liveries. I would then manually add it to every texture adjusting the intensity using the alpha channel.

**Comment by gilbertohasnofb** [comment]
Another advantage of overlays is that if I ever alter the xcf files by adding more stuff, changing a texture, etc. we don't have to bother to bake them again, as this will be a layer on top of everything else.

**Comment by wlbragg** [comment]
@gilbertohasnofb this is a first attempt. It has issues. I think there is some overlapping UV texturing. Also I need to know exactly which parts or objects we want to include in the AO overlay. There are settings we may be able to adjust to get better results as well. In order to produce this it requires appending the objects I want to include in the bake to a new blend so I can control things better and eliminate all the extra garbage we don't need. or want. It's going to take some experimenting.
Let me know you impression of this first try. Maybe you can bring it into Gimp and overlay it to test what it's going to look like. Then give me your impressions and feedback for the next attempt.
![ao](https://user-images.githubusercontent.com/11069680/39910355-8af5d0ea-54bc-11e8-9075-266c3b8c867c.png)

**Comment by wlbragg** [comment]
Also, what size do you want this?

**Comment by gilbertohasnofb** [comment]
The size should be 2048 x 2048. But there are some weird things in this image above. Is the GPS contributing to this AOMap? If so, we should hide it. Also, the place where the trim needle is looks way too dark, though we do need the window of the needle to be darker (similarly to how the fuel gauge looks like in the image above). There are a lot of artifacts around some of the gauge circles, some of them looks like the barrel of a gun with some weird lines. I also have the impression that the back side of the pedestal should have considerable more shadows to it. And last, it looks like the AOmap is 'spilling out' of objects, is that normal? That is, they are larger than the place they should cover in the texture, sometimes overlapping (look at the bottom section of the pedestal for instance, many items are overlapping above)

**Comment by wlbragg** [comment]
>Is the GPS contributing to this AOMap? 

No, that is a remnant from trying to fix the panel behind the radio stack I think. If so, I will be able to clean that artifact up.

>There are a lot of artifacts around some of the gauge circles, some of them looks like the barrel of a gun with some weird lines.

If your referring to what I think you might be, it could be a anomaly in the mesh itself. Every hole has a raised bump on it. I'm wondering if that was for a recessed bulb for instrument lighting of some sort VS back lighting.

![bump](https://user-images.githubusercontent.com/11069680/39934666-211578ce-550c-11e8-82bc-9e0ffe33b309.jpg)

> it looks like the AOmap is 'spilling out' of objects, is that normal?

I think that is the overlap setting, I'm not really sure why it is always suggested to us a reasonably large number for that. I think so you get the map solid to the edges? But that would interfere with two objects butting up to each other. We should be able to adjust that.

So this is close to what we want, barring the comments above? Are we missing anything else that should be included?

**Comment by legoboyvdlp** [comment]
![image](https://user-images.githubusercontent.com/10434911/39935236-3c43c48c-5540-11e8-8836-ec2256f6604e.png)

An attempt at making those bumps?

From the side there is no bump visible:

![image](https://user-images.githubusercontent.com/10434911/39935281-67c2520e-5540-11e8-85b8-6a2d55f4973e.png)

**Comment by legoboyvdlp** [comment]
Actually, a look at  a FSX product suggests it might actually be a bulb:

![image](https://user-images.githubusercontent.com/10434911/39935327-8f09579a-5540-11e8-9def-c4b2e6e4c395.png)

Note how the light shines from the upper left.

**Comment by gilbertohasnofb** [comment]
> If your referring to what I think you might be, it could be a anomaly in the mesh itself. Every hole has a raised bump on it. I'm wondering if that was for a recessed bulb for instrument lighting of some sort VS back lighting.

No, that's not an anomaly, the model P does have those bumps. I am referring to these:

![image](https://user-images.githubusercontent.com/5700795/39935329-8fd78778-5540-11e8-90fe-8d8605439e36.png)

> So this is close to what we want, barring the comments above? Are we missing anything else that should be included?

Yep, we are close. With these sort of images I can improve the textures a lot. And I would need a map for each of the three interior images.

**Comment by gilbertohasnofb** [comment]
About the bump, they do change from panel to panel and model to model. Ours look fine, very close to the OH-CTL. Let's not touch them.

![aad](https://user-images.githubusercontent.com/5700795/39935409-d3450e72-5540-11e8-8e53-88f73e9a3034.jpg)

**Comment by wlbragg** [comment]
>Also, the place where the trim needle is looks way too dark, though we do need the window of the needle to be darker 

You may be looking at the wrong thing, this should clarify it.
![ao1](https://user-images.githubusercontent.com/11069680/39935684-76c6a33a-550f-11e8-963c-33357864a506.jpg)
![ao2](https://user-images.githubusercontent.com/11069680/39935688-78628c72-550f-11e8-8f4d-1abb5c4d4d02.jpg)

**Comment by gilbertohasnofb** [comment]
You are right, I was looking at the wrong thing. Still. that window needs to be darker and that black square shouldn't be there. Also note that I can make manual adjustments if we can't make something come out of Blender, e.g. the trim needle window can easily be created in GIMP using the reference above. But I will only start to manually improve these once we are absolutely sure that the AOMap is fine, else another export from Blender would make me lose every modification. Makes sense?

**Comment by wlbragg** [comment]
>Still. that window needs to be darker 

Same answer, I don't think I have that kind of control over Blender, it's going to interpret it how it sees it. You may have to manually adjust these  things. Although I am nit sure I agree that it needs to be any darker (if were talking about the same thing, the trim needle recess). Shouldn't this stuff be pretty subtle so as to blend with most lighting conditions. Think of what that same area will look like with direct sun light on it (I know not totally likely, but for arguments sake).

>black square shouldn't be there.

That stuff you will have to edit out. Blender sees mesh with other mesh over it it is going to make it black (as in covered).

and that black square shouldn't be there.

**Comment by gilbertohasnofb** [comment]
Ok, I will try to do all this using GIMP. But could you try checking why those artifacts I pointed out are there?

**Comment by wlbragg** [comment]
>why those artifacts I pointed out are there?

I'm running another bake at 2048 with adjustment we can make, as we speak. Those "artifacts" were caused by the "smoothness" angle setting. It was at 180 deg. Reducing it to 90 removes the artifacts. It's possible that will create other artifacts somewhere else, we will have to wait and see.

**Comment by wlbragg** [comment]
I added the interior chassis to the blend to see if that will darken areas you want to see darker. So far I can't tell for sure but it looks like it may be affecting it.

**Comment by gilbertohasnofb** [comment]
Great! If the new map will still have artifacts then please bake me another one with 180 deg of smoothness angle, so that I can make a patch work and copy and paste the clean parts from one to the other.

If you have a chance, try generating the other two maps for the other interior textures.

**Comment by wlbragg** [comment]
The margin in the first bake 1024x1024 was 16, I reduced it to 6 for this 2048x2048 so hopefully that will be enough. I don't know how far we can go down in margin before it causes trouble on the edges, if it does at all. I'm only going by what I have read on the subject that they always say to leave a decent margin, whatever that means?

**Comment by gilbertohasnofb** [comment]
No problems, let's see how this ones comes out.

**Comment by gilbertohasnofb** [comment]
This looks much better, but I still do see those artifacts around the gauges. That's my only gripe with this.

**Comment by gilbertohasnofb** [comment]
Otherwise looks excellent! Then I can tweak the strength of certain areas manually in GIMP :smile:

**Comment by wlbragg** [comment]
>I still do see those artifacts around the gauges

There is a setting in the bake that is possibly able to take that out. Also I did this at 64 samples VS 128 which I did last time. I think that will only affect the graininess and not that type of artifact.
There is a many factors that can be adjusted in the bake. The increase in darkness was caused by adding the chassis to the blend. I could darken it using blender bake and gathering settings instead of adding the chassis if this is too dark?

**Comment by gilbertohasnofb** [comment]
It's not too dark, it's fine. The more dynamic range between the dark and light parts the better for me, I can then control this with opacity in GIMP. If the dynamic range is too low (e.g. all too light) then I will have a hard time expanding it when necessary.

**Comment by wlbragg** [comment]
@gilbertohasnofb 
There is a bias setting that takes out the artifact but it changes the overall bake as well. This was a bias of .15 (recommended for taking out artifacts in a sphere).
Try it with somewhat less of a bias?
![ao](https://user-images.githubusercontent.com/11069680/39940395-cd04d9a6-551e-11e8-94b6-7ea8992f3712.png)

**Comment by wlbragg** [comment]
I also reduced sampling size (graininess) considerably (faster bake).

**Comment by wlbragg** [comment]
I'm running it again with the previous sampling size and about half the bias,

**Comment by wlbragg** [comment]
Hum, not much difference. I'll try one more run with the bias cut even more.
![ao1](https://user-images.githubusercontent.com/11069680/39942577-7a5df0a0-5525-11e8-9690-45130df2b231.png)

**Comment by wlbragg** [comment]
Hum I didn't get the size change.
This is using approximation instead of ray tracing, the correct size.
![aoappx](https://user-images.githubusercontent.com/11069680/39943477-8d31bab0-5528-11e8-80b8-6c21c31e000b.png)
I'll do one more with ray tracing, and a low bias for artifact and you can decide which you like best.

**Comment by wlbragg** [comment]
Ray-tracing, .001 bias 2048x
![ao](https://user-images.githubusercontent.com/11069680/39944322-db66128c-552b-11e8-9a39-7aa4ebf7bf70.png)

**Comment by gilbertohasnofb** [comment]
@wlbragg most of these new ones seem inverted, e.g. look at the pedestal parts and compare with the first one [here](https://github.com/c172p-team/c172p-detailed/issues/798#issuecomment-388429795). I think that first one was the best by far and the one with most dynamic range which is important for me. I just wish those artifacts would be a bit less pronounced and that the black square in the pedestal would disappear.

**Comment by gilbertohasnofb** [comment]
If this is becoming too much work for you I can also try to fix those manually, maybe I can blur those artifacts and possibly when they are applied as a thin overlay that wont be noticeable.

**Comment by gilbertohasnofb** [comment]
Great, no rush at all! We can review it as many times as it takes until we get it right :+1:

**Comment by wlbragg** [comment]
I think this is about as good as I am going to get this. What do you think @gilbertohasnofb? You'll have to edit the black areas.
![ao](https://user-images.githubusercontent.com/11069680/39953306-7358299e-556e-11e8-9725-fb7f9cb0b190.png)

**Comment by legoboyvdlp** [comment]
Doesn't the texture, lightmap, and AO map all have to be on the same uvmap? i.e. any changes need to be done to the original as well?

**Comment by gilbertohasnofb** [comment]
Not bad, the panel looks fine and totally workable but panel_parts look a bit more messy. For instance, it's clear that the lighter is mapped to the a place it shouldn't since you were mapping it for testing to an existing texture before I pushed my new one. And what about the interior aomap?

**Comment by gilbertohasnofb** [comment]
@wlbragg one more thing: I was hoping that the glareshield would make some shadow on the top of the panel in the AO map, do you know why that isn't the case?

**Comment by wlbragg** [comment]
>panel_parts look a bit more messy.

This was just a first run to establish what's there and what is available for our use. I guess it would help to know specifics as to what you want out of it? if there are objects that are of no use I can eliminate them from the map before baking so you don't have to cut them. 

>the lighter is mapped to the a place it shouldn't since you were mapping it for testing to an existing texture before I pushed my new one

Yeah, I'll burn it again. First if you want to mark up anything you want removed or anything else you want to comment on, go ahead.

> And what about the interior aomap?

I guess I should assemble the mesh involved, burn it and see what we get. I hadn't got to it yet.

>Doesn't the texture, lightmap, and AO map all have to be on the same uvmap? i.e. any changes need to be done to the original as well?

@legoboyvdlp my procedure for creating the ao map is to append to a new blend only the parts I want to include in the map and join them. I also append any light altering parts not a part of the map that will have an effect on it. I remove all textures and material settings. Make a new "all white" material and then bake it. I can repeat that process for any other texture sheets used in the interior and add those in the same blend and bake it all at once and get multiple ao maps, one for each individual texture sheet.
I'm guilty of reusing resources too often in on a texture (mapping multiple objects to the same texture). You just can't be doing that if you ever plan to eventually make ao or light maps.

For the light map I would do something similar as far as the assembly. I've never had the opportunity to  worked with a blend as well organized as say the c182s, now that is a well organized blend.
But to answer your question, yes, kind of.  The ao is independent in so much as @gilbertohasnofb is going to simply overlay the parts he wants to use over the texture. Then when it comes time for the light maps, we will organize an optimized blend with lights of appropriate strength and position placed and  to the necessary bakes. I think for the most part the c172p uv maps are fairly sound and we should be able to produce some decent results with possibly a few modifications.

The last interior ao map bake I have to do is the last piece to the puzzle as far as I am concerned. As soon as I see the results (or simply look at the entire UV map a little closer), I'll have a good understanding of what we will need to do to fix it.
For example, I  can tell you right now, if we want the interior doors included in an ao map we might be able to cheat an get a bake that will work for both as they are mapped to the same space. But that isn't going to work for certain light maps that might affect one door more or less than the other. If we plan to do that then we need to rework the texture sheet right now and give each door it's own uv space.

**Comment by wlbragg** [comment]
>I was hoping that the glareshield would make some shadow on the top of the panel in the AO map, do you know why that isn't the case?

@gilbertohasnofb yep, I forgot to include it in the bake. I'll add it and re-bake.

**Comment by gilbertohasnofb** [comment]
>  if there are objects that are of no use I can eliminate them from the map before baking so you don't have to cut them.

Just keep them all, it's easier for me to cut them out in GIMP than to change my mind later and need you to bake a new one.

**Comment by wlbragg** [comment]
@gilbertohasnofb OK, here's the deal. I'm my rush to improve stuff I have taken a few shortcuts when uv-mapping some of the exterior object (mostly shiny stuff like antennas or chrome parts like beacon mount, etc.). I mapped some of that stuff to the panel_parts.png instead of having you create new textures. It has now caught up to me. If we ever hope to make interior light maps, or even be able to use the ao maps for some the interior stuff we need to separate those objects uv mapping to their own uv space. Either on a new texture sheet, a different texture sheet or even the same texture sheet but in their own space.
Ideally we map the exterior stuff to a separate "new" sheet or add it to fuselage.png.
How would you like to proceed? You want me to start producing a graphical example list of the exterior objects I am referring to and their current mappings so you can get an idea of what to create for their own texture space?
Sorry about this, I don't think it is going to be "extensive". In my defense (I know it would hold up in court :fearful: ) I really didn't think about light maps or even ao maps and I guess I didn't think that particular texture would be one you might want to improve with ao. Had I given it the slightest thought I would have at least realized that if we ever wanted to do light maps I shouldn't do what I did. Please forgive me, another lesson learned, I'm not happy unless I learn the hard way.
I can still give you the panel ao map. I'm finishing up the final arrangement of object that should influence the AO on it.

**Comment by wlbragg** [comment]
@gilbertohasnofb here is the final map with everything on it. Anything black more than likely has another object over it or it is in deep shadow.
![ao](https://user-images.githubusercontent.com/11069680/39961015-3190c662-55f3-11e8-86b8-4a0f8097165a.png)

**Comment by wlbragg** [comment]
Here is the current panel_parts.png ao map. This is the one we need to pull some of the exterior uv mapping off of. That is why some of the objects don't look right.
![flap](https://user-images.githubusercontent.com/11069680/39961489-a89f242a-55fc-11e8-8b83-0b7ef9262e52.jpg)
![aop](https://user-images.githubusercontent.com/11069680/39961490-a8c08124-55fc-11e8-9895-014d4200bed8.png)

Fuel selector in case you want it.
![aofuel](https://user-images.githubusercontent.com/11069680/39961493-c4d0ee8a-55fc-11e8-8807-e2d98c635acf.png)

**Comment by gilbertohasnofb** [comment]
> OK, here's the deal. I'm my rush to improve stuff I have taken a few shortcuts when uv-mapping some of the exterior object (mostly shiny stuff like antennas or chrome parts like beacon mount, etc.). I mapped some of that stuff to the panel_parts.png instead of having you create new textures.

@wlbragg First of all, not a bit deal, we can clean this up with a small amount of effort, so no feeling bad about that :smile: Well, I actually imagined that was the case when I saw those dark patches in the AOMap. Do note that I absolutely never ever mind making new textures and we have plenty of empty space in nearly all texture files, so it's always good to have one texture per object. Also, this way I can always alter and improve textures without fearing that they might have consequences in a different part, so it's not only a problem with light and AO maps.

> Ideally we map the exterior stuff to a separate "new" sheet or add it to fuselage.png.

I will add whatever you need to `fuselage.png`, could you please tell me what are the things that need to be added there?

> Sorry about this, I don't think it is going to be "extensive".

Me neither, but let's make sure we fix this properly. Let's not leave a single texture double mapped, ok?

> Here is the current panel_parts.png ao map. This is the one we need to pull some of the exterior uv mapping off of. That is why some of the objects don't look right.

Notice that the lighter is still mapped to the wrong place.

Also, a question about panel_parts: can you confirm to me that we don't use that bronze coin texture on the right side? I think it was thevirtualfer that had this idea of adding some coins to the floor, but I really disliked it. If we don't use it then I will make space in the texture so future stuff can go there. If we do have the coins on the floor then I would suggest deleting them.

I think we don't need a map for the fuel selector but I am dying to see the map for the interior cabin, pleeeeease bake me that :rofl:

**Comment by wlbragg** [comment]
@gilbertohasnofb 

>Notice that the lighter is still mapped to the wrong place.

That's been corrected and will show up on the next bake.

>can you confirm to me that we don't use that bronze coin texture on the right side? 

Yes, that was removed long ago.

**Comment by wlbragg** [comment]
@gilbertohasnofb following is a list of items that will have to be dealt with at some point if we want light maps and good AO mapping.
I'll start with the exterior stuff that needs moved off panel_parts.png.
![ant1a](https://user-images.githubusercontent.com/11069680/39964240-dc9a5fee-5645-11e8-97a1-8d5bf9504944.jpg)
![ant1](https://user-images.githubusercontent.com/11069680/39964246-fb62143a-5645-11e8-86d7-c8a67877a3ec.jpg)
![ant2](https://user-images.githubusercontent.com/11069680/39964248-0ed157b0-5646-11e8-94aa-523b857f28cb.jpg)
![ant3](https://user-images.githubusercontent.com/11069680/39964249-11bbe76a-5646-11e8-89bd-c59094e35850.jpg)
![ant4](https://user-images.githubusercontent.com/11069680/39964252-17056b42-5646-11e8-9485-f8e612d4d3d4.jpg)
![ant4a](https://user-images.githubusercontent.com/11069680/39964253-19059f48-5646-11e8-8447-52bdbe217ca8.jpg)

Below starts items that need their own texture space so the light maps will work properly. This first one is special as the interior metal ring on the glass (x2) should be mapped to the panel_parts.png texture so they can pick up the appropriate light for the light maps. But the outside rings (x2) on the glass should be mapped to fuselage.png do they aren't part of the interior and thus don't pick up in the light maps.
![ant5](https://user-images.githubusercontent.com/11069680/39964282-bbe26c0a-5646-11e8-8f69-b996a1d8810f.jpg)
![ant6](https://user-images.githubusercontent.com/11069680/39964286-dfb35e00-5646-11e8-8b46-4e5e89c75a46.jpg)

The window handles and the next images below of the door handles should be mapped to different textures for left and right doors and windows as different light sources will affect each one differently, so they each need their own texture space.
![ant7](https://user-images.githubusercontent.com/11069680/39964300-2ca19a38-5647-11e8-9f03-272cda459db2.jpg)

Any knobs as below that are mapped to the same texture space need to be remapped and provided their own separate texture space do the light maps work correctly..
![ant8](https://user-images.githubusercontent.com/11069680/39964308-6d742300-5647-11e8-9f58-2fabd2478def.jpg)

And last, every breaker needs it's own separate graphic, every switch needs it's own separate graphic . Upper right hand corner, that knob base needs it's own texture and space. Also look at the lower right (red circle around the dark texture), now this type of mapping will work if I remove any overlapping parts as each uv mapped object actually has its own space here. At least it will once I separate them by moving them apart a little bit.
![map1](https://user-images.githubusercontent.com/11069680/39964347-2cb28b4e-5648-11e8-93a5-790c3f36ccbb.jpg)

Note: there will be a few things on the other interior texture ao map (the one you anxiously waiting for) that will need to be separated and mapped to their own space, such as the doors. Right now both the doors are mapped to the same space. I need two doors to map each one separately. I may still bake you a test ao of that texture sheet om the morning as I need to organize it anyway. But you wont get a final of either of the last two AO maps until we separate all the objects into their own uv space.

**Comment by gilbertohasnofb** [comment]
@wlbragg Great, I will work on this soon. Shouldn't be too bad, I just hope we will have enough space for everything in this texture file! We have many switches and circuit breakers so that might be a problem. Do you think that if did not multiply those for each switch and breaker they would look too bad in a light map? Or could we possibly get away with this? The light in an old 172P will be quite minimal I believe, see that other issue for the discussion.

**Comment by gilbertohasnofb** [comment]
@wlbragg I opened a new issue for this, see: https://github.com/c172p-team/c172p-detailed/issues/1104

**Comment by wlbragg** [comment]
>Do you think that if did not multiply those for each switch and breaker they would look too bad in a light map?

They have to be separated to their own space. If you look at the shuttle you'd be surprised how much you can cram in a space and how small some of these textures can be yet still look good. The switches in particular could be half the size and still be plenty big and of high enough resolution to look good. The breakers size could be reduce a little and still look good. You could get most of the room you need for them simply by resizing them. That square blank white space that the four breakers occupy right now, I think if you reduced them enough to place two rows tall and at least 6 wide, that only leaves two more to find space for.
When I look at our texture sheet I see plenty of white space, even if it requires rearranging existing mappings to get the space back that'll be OK. This type of mapping is easy for me to move so we don't need to worry about complexity of the mapping having to move. I thing there are several textures that are much bigger than they need to be. Even the new texture for the mike_mount could easily be half the size and still look good.
For example, this is what I am talking about. Most of our textures are bigger than necessary to achieve the same good results. in addition to the markup below, every knob could easily be reduced in size and grouped closer together. The door handle could easily be half the size to fit two. I think once you look at it with organization in mind, I'm sure you'll see all the free space as I do.
![map](https://user-images.githubusercontent.com/11069680/39969229-80a42ffe-569e-11e8-8c87-aa436bc1bf3f.jpg)

**Comment by wlbragg** [comment]
@gilbertohasnofb here is the first pass at interior.png. This is going to be the tough one to separate as the doors are of a substantial size.
Also see #1104 for the current mapping of everything on that sheet. 
![aoint](https://user-images.githubusercontent.com/11069680/39970618-3d0961a4-56b4-11e8-8053-79e79aa1c064.png)

**Comment by gilbertohasnofb** [comment]
@wlbragg why is it so dark? Could we tweak it a bit to add more contrast? Also, a couple of things: the metal panel in the back of the interior is missing from the map, and there are a couple of weird artifacts here and there, most of which I can clean except for the ones in the seats which look quite problematic.

**Comment by wlbragg** [comment]
@gilbertohasnofb I noticed the darkness, I will investigate that. The imperfections in the seat are probably multi-objects mapped to them, I'll investigate that as well.

>the metal panel in the back of the interior is missing from the map
Hum, let me see what I did with it!

**Comment by wlbragg** [comment]
Final panel.png AO map?
I notice it looks darker, but I have added more objects such as seats flooring, firewalls, etc., which may account for it getting darker?
I figure you can adjust in Gimp if needed?
![aopanel](https://user-images.githubusercontent.com/11069680/40012479-707576ec-5770-11e8-844f-1da2e39c1dbe.png)

**Comment by gilbertohasnofb** [comment]
@wlbragg The problem is that what I am interested the most is in having a natural looking shadow from the AO map in the pedestal, and if the pedestal is nearly completely black then there is not much use to me. Would it be too much trouble to try doing this again but removing some of the interior objects? That influences the pedestal the most will be the panel and the glareshield (in case the bottom of the panel is hollow, which I guess it might be).

**Comment by wlbragg** [comment]
I found the correct setting to adjust, These are all low resolution. Choose your option or anything in between these and I can bake a high res final.
![aopanel-75](https://user-images.githubusercontent.com/11069680/40013035-179c9a9e-5772-11e8-8ee8-a973b6e6b6eb.png)
![aopanel-5](https://user-images.githubusercontent.com/11069680/40013044-1a143818-5772-11e8-8dd5-abb1eebb627b.png)
![aopanel-05](https://user-images.githubusercontent.com/11069680/40013047-1c7261a2-5772-11e8-91ca-3e259765928c.png)

**Comment by gilbertohasnofb** [comment]
If it's not too much trouble, could you send me both no 1 and no 2 from your last post?

**Comment by wlbragg** [comment]
I was baking this "hi-resolution" ao map while I was away. We will have to do it again after i re-map the pedestal, but for now this gives you yet another test for contrast.
Just so you know. The two you are requesting are distance setting of 1. and .75.
This one below is at .875. As soon as I remap the pedestal I will bake you another one or two "final" versions. In the mean time, please decide what settings you want for as many bakes as you want.
Remember, your last request was one at 1. and one at .75. Below is one at .875 in a final version resolution other than it doesn't contain the coming soon re-map of the pedestal.

![aopanel-875](https://user-images.githubusercontent.com/11069680/40022375-6c120abc-578d-11e8-8775-a9bcec0ac3db.png)

**Comment by wlbragg** [comment]
@gilbertohasnofb keep in mind, it looks like the higher resolution is baking darker than the three examples above when you request what setting to use..

**Comment by gilbertohasnofb** [comment]
@wlbragg Yes, the last one is definitely too dark. Could you try two different values lower than .875 so that we can try to get something like the first and second ones from your previous post?

**Comment by gilbertohasnofb** [comment]
These look excellent, thank you so much. When you have a chance please do bake me the interior one as well and then we will be done with this!

**Comment by gilbertohasnofb** [comment]
@wlbragg Also, could I ask you for a mega favour? Could you bake me another AO map for panel_parts but including the glareshield this time? This texture is actually 1024 x 1024, only interior and panel are 2048 x 2048 by the way.

**Comment by gilbertohasnofb** [comment]
And these are the results so far:

![fg-2](https://user-images.githubusercontent.com/5700795/40080090-d72b8b08-5881-11e8-9738-dbb1d77b61d6.jpg)
![fg-3](https://user-images.githubusercontent.com/5700795/40080091-d7655b4e-5881-11e8-9bbf-e36803e06efb.jpg)

Here is a before and after of a lighter livery (which was done just as a test and is not pushed, as agreed I will work only on the default one until this is all set and done. Also notice that the knobs are wrongly mapped when showing the old texture due to the changes we did to the UV map):

before/after:
![fg-4](https://user-images.githubusercontent.com/5700795/40080155-0de448ba-5882-11e8-8a5b-c8b45c121558.jpg)
![fg-5](https://user-images.githubusercontent.com/5700795/40080156-0e045484-5882-11e8-8f2e-aed982ab326d.jpg)

before/after:
![fg-6](https://user-images.githubusercontent.com/5700795/40080173-1a7bf5f0-5882-11e8-8a12-b54d0cf0dbc7.jpg)
![fg-7](https://user-images.githubusercontent.com/5700795/40080174-1aacc144-5882-11e8-81c4-c903865f400e.jpg)

I might need to tone down the effect a tiny bit for the light panels, but that will be done on a livery to livery basis so please excuse the contrast being too much in the images above. @wlbragg what's your impression? I find the result extremely convincing! :smile:

**Comment by gilbertohasnofb** [comment]
Here is a before and after of the black panel, the difference is also huge:

![fg-0](https://user-images.githubusercontent.com/5700795/40080576-4b27ee92-5883-11e8-86f2-c6407b81b70c.jpg)
![fg-1](https://user-images.githubusercontent.com/5700795/40080598-5457ba1a-5883-11e8-9587-5901bd8d11c1.jpg)

**Comment by gilbertohasnofb** [comment]
Sorry but one more shot, the pedestal is looking so good now!

![fg-3](https://user-images.githubusercontent.com/5700795/40081121-fa8aed34-5884-11e8-9388-ca4ed2f180f7.jpg)

@wlbragg I found a white line in between the pedestal and the back panel behind the pedals, could you check what is causing that?

![fg-2](https://user-images.githubusercontent.com/5700795/40081150-0bf9318e-5885-11e8-803e-08c38c826e0e.jpg)

**Comment by gilbertohasnofb** [comment]
Sorry, two more shots. I think this is looking _really_ good!

![fg-0](https://user-images.githubusercontent.com/5700795/40085105-4f1f78a4-5891-11e8-801b-7e27debab5fd.jpg)
![fg-1](https://user-images.githubusercontent.com/5700795/40085106-4f3f8eaa-5891-11e8-8384-be7ef630a371.jpg)

**Comment by wlbragg** [comment]
@gilbertohasnofb 
>@wlbragg what's your impression? I find the result extremely convincing!

Yes looks great.

>Also notice that the knobs are wrongly mapped when showing the old texture due to the changes we did to the UV map)

You are aware that will be corrected when you update liveries, right?

Could you bake me another AO map for panel_parts but including the glareshield this time?

Even though you did it manually I am baking everything at least one more time. I'm almost done. Why am I baking them again you may ask, I cleaned a bunch of stuff up and made the maps a bit more useful. I think you'll see what all I am talking about when you see them. Quite time consuming work however with the double mapping (mirroring) we have on the interior.png and the couple things we still have double mapped on the panel_parts.png (window latches, door handles. The panel.png was the most time consuming as it had multiple mirrored mapping issues, like chassis, doors, seats (including the back seat). I think you'll be happy with the results. Also I found an hidden issue with the door arm rests and the handles, somewhere along the line they got recessed into the door and I pulled them back out and re-centered the handle animations. Also I didn't forget to add the glare shield to the bake, although I was surprised how little it was influencing the panel this time around.
![doorh3](https://user-images.githubusercontent.com/11069680/40139524-bfa78ee8-5915-11e8-810a-213114c328b9.jpg)
![doorh2](https://user-images.githubusercontent.com/11069680/40139526-c352ae42-5915-11e8-9c2a-4eb64574a26c.jpg)
![doorh](https://user-images.githubusercontent.com/11069680/40139529-c5678a22-5915-11e8-8b16-9273c22f0f84.jpg)
Anyway, I am on the last bake's (panel_parts.png) although it looks like I am going to have to do them again as I am still seeing some artifacts I can't account for.
@gilbertohasnofb I take it you cutting and pasting and mixing and matching your own AO overlay? If so then I am only going to bake what I maps I think you may still be missing clean parts from, Like the door handles interior.png of course VS baking everything yet another time. If you are using them as is then I would still like to bake a final product. But if you've been cutting the best stuff out and mixing it all up then there is no need to do them all over, Your call!
Here is interior.png and panel.png
![aoint 25](https://user-images.githubusercontent.com/11069680/40139826-c05f9a0a-5916-11e8-8ebe-7269762a6226.png)
![aoint 5](https://user-images.githubusercontent.com/11069680/40139830-c1ac9c6e-5916-11e8-8e24-923419638b20.png)
![aopanel 25](https://user-images.githubusercontent.com/11069680/40139833-c442919a-5916-11e8-8381-fe00c5258f45.png)
![aopanel5](https://user-images.githubusercontent.com/11069680/40139834-c5a775c8-5916-11e8-92bf-411a9b7dde4d.png)
Afew more things to fix on panel_parts.png and you will still need another pane.png once the glovebox is finished.

**Comment by wlbragg** [comment]
@gilbertohasnofb these two small items are why am am baking yet another panel_parts.png AO, every run I discover more issues that really should be dealt with in time to make this easier and to improve the visual.
These two object textures could be improved and the ao would be so much better because the visible undersides would be darker. In this instance we can get away with it because I think the AO we get that is also going to be applied to the double mapped underside and end won't be in a critical edge area between dark and light. In fact, the underside of the handle when animated if it were to have a dark ao applied to it (because it is hidden from light when baked) would look wrong when animated to an open position. But the end and the window handle visual could be vastly improved with a better texture. What I am doing is to remove those duplicate faces from the AO blend so they don't influence the bake.
![dhand](https://user-images.githubusercontent.com/11069680/40141396-73474f56-591b-11e8-85b4-fba1b530cb4d.jpg)
![whand](https://user-images.githubusercontent.com/11069680/40141398-756f7b64-591b-11e8-990e-0a23c83bbf2c.jpg)

**Comment by wlbragg** [comment]
@gilbertohasnofb  the window handle might be worth it. The door handle I'm not so sure about as far as baking the ao map is concerned. reason being, you almost need to make the AO map for the handle by hand as it is going to make black because it is hidden from light when closed. You will want to make the AO in the shadow areas only for in the open state so it looks right when open, when closed that portion is all hidden anyways.

**Comment by wlbragg** [comment]
@gilbertohasnofb here is panel_parts.png at .5. I am baking the one more at .25. It should be done in 30 minutes or so.

**Comment by wlbragg** [comment]
Oops, still forgot the glare shield vent in the main blend and the AO blends.

**Comment by wlbragg** [comment]
panel_parts.png at .5
![aopanelparts 5](https://user-images.githubusercontent.com/11069680/40147210-73f34406-592e-11e8-8c06-7756657ad837.png)

**Comment by gilbertohasnofb** [comment]
>> Also notice that the knobs are wrongly mapped when showing the old texture due to the changes we did to the UV map)
>
> You are aware that will be corrected when you update liveries, right?

Yep, I am aware of that, I jsut informed that to avoid anyone being confused at the screenshots.

> Even though you did it manually I am baking everything at least one more time. I'm almost done. Why am I baking them again you may ask, I cleaned a bunch of stuff up and made the maps a bit more useful. I think you'll see what all I am talking about when you see them

Great, I will take a look at this tomorrow and incorporate whatever changes might be useful. Really appreciate all the work you are doing!  

> Also I found an hidden issue with the door arm rests and the handles, somewhere along the line they got recessed into the door and I pulled them back out and re-centered the handle animations. Also I didn't forget to add the glare shield to the bake, although I was surprised how little it was influencing the panel this time around.

Thansk for fixing the arm rests and about the glare shield I have manually created an 'AO map' myself which is looking pretty decent.

> Anyway, I am on the last bake's (panel_parts.png) although it looks like I am going to have to do them again as I am still seeing some artifacts I can't account for.

I've clean everything manually and to be honest I am not in need of a new panel_parts AO map, that one is looking quite nice.

> I take it you cutting and pasting and mixing and matching your own AO overlay? 

Yep, that works for me

> your pictures aren't doing it justice. It is absolutely phenomenal work. Looks incredible in the sim. Well worth all the effort.

Thank you so much! I am also super happy with the results so far :smile:

**Comment by gilbertohasnofb** [comment]
So here is my plan for tomorrow:

* inspect if the window handle and door handle are worth to be broken into two textures which you would need to re-UV-map and re-bake panel_parts
* work on `interior.png` and add the AO map to it
* merge your new AO maps to the current ones in `panel` and `panel_parts` in case they change something
* once all is good and we agree that the result is fine I will then start working on all other textures

@wlbragg sounds reasonable to you?

**Comment by wlbragg** [comment]
>sounds reasonable to you?

@gilbertohasnofb Yep, here is the last panel_parts at .25
![aopanel 25](https://user-images.githubusercontent.com/11069680/40148668-d2cf54e6-5934-11e8-99b5-004f6d86ee7e.png)

**Comment by gilbertohasnofb** [comment]
@wlbragg Ok, so I pushed everything. The window handle has been duplicated and needs re-uv-map, the door handle was an overkill and I left it alone. I added the AO map to `interior` and I improved the seat texture (less leathery, more uniform). `panel` and `panel_parts` have had a minor updated to the AO map using parts of what you created above. There is a new texture for the glove compartment (see my other comment in that issue).  

Let me know what you think. We might need only a 1024 x 1024 AO map for `panel_parts` since we have a new texture for the window handle and another for `panel` due to the new glove compartiment, all else looks good to me.

**Comment by wlbragg** [comment]
>We might need only a 1024 x 1024 AO map for panel_parts

Ah, I didn't realize panel_parts was 1024X, will do.

Good thanks, should be fall finished by tomorrow late.

**Comment by onox** [comment]
@gilbertohasnofb It looks like the new vintage livery doesn't have the AO map yet?

**Comment by gilbertohasnofb** [comment]
@onox only the default livery has the AO map so far (I posted a couple of shots of ZS-MDF but I did not push those). It's too much work to alter things on all liveries at the same time, so @wlbragg and I agreed to work only on 4X-CHV and once we agreed that all looks good I will then apply the same to all other liveries.

**Comment by wlbragg** [comment]
>what is this thing in the AO map? There is no texture on that place, it's all blank:
@gilbertohasnofb that is the ElevatorTrimPos mapped to a blank white area.

**Comment by wlbragg** [comment]
>you need to tell me about these things else for me it's all blank space

Sorry, it's something I moved recently because I noticed it mapped somewhere else odd. I see it in Blender but don't think to remember in Gimp it wouldn't even be there at all.

Thing look really good overall. 

I wasn't 100% satisfied with the window latches. I don't like the stripes from the edge mapping so I moved the edge mapping a little, just enough to remove those stripes.

With that I think your good to proceed.

![latch](https://user-images.githubusercontent.com/11069680/40201449-b6255d9e-59e4-11e8-910e-8c177a3f72c3.jpg)

**Comment by gilbertohasnofb** [comment]
> Sorry, it's something I moved recently because I noticed it mapped somewhere else odd. I see it in Blender but don't think to remember in Gimp it wouldn't even be there at all.

No problems! Let's just always try to remember to keep communication between textures and 3D model, and I'm always happy to create new ones as well so no need to duplicate anything :smile: :+1: 

> I wasn't 100% satisfied with the window latches. I don't like the stripes from the edge mapping so I moved the edge mapping a little, just enough to remove those stripes.

Thanks for fixing that! I did notice it but I forgot to remind you about that.

> With that I think your good to proceed.

Great, I will try to get this done maybe still today. Do check my reports about some bugs/issues on the PR page.

> Thing look really good overall.

Yes it does! I'm quite happy with the results as well :smile:

**Comment by gilbertohasnofb** [comment]
All liveries are pushed now! :smile:
